### PR TITLE
Support Unicode file paths when loading an ONNX file

### LINKED ISSUE #5761: Support Unicode characters in proto paths
# Bug Report

### Is the issue related to model conversion?

No
### Describe the bug

1. the onnx name is simple   Chinese，such as  **“士大夫.onnx”** 
2. ```
     output_file_path = 'D:\\士大夫.onnx'
     onnx.checker.check_model(output_file_path)```
3. after run ，throw  **ValidationError**
```
   onnx.onnx_cpp2py_export.checker.ValidationError: Unable to open proto file: D:\士大夫.onnx. Please check if it is a valid proto. 
 [fail_check("Unable to open proto file: ", proto_path, ". Please check if it is a valid proto. ");]
```

https://github.com/onnx/onnx/blob/b60f69412abb5393ab819b936b473f83867f6c87/onnx/common/file_utils.h#L21

### System information

Windows
onnx version 1.15.0

### PR DESCRIPTION
**Pull request by captainIT**
### Description

Support Unicode file paths when loading an ONNX file.

### Motivation and Context

Fix #5761

### CODE REVIEW

**Comment by captainIT** [comment]
fix #5761

**Review by justinchuby — APPROVED** [review]

File: onnx/common/file_utils.h [hunk 1]
```diff
@@ -16,7 +16,8 @@ namespace ONNX_NAMESPACE {
 
 template <typename T>
 void LoadProtoFromPath(const std::string proto_path, T& proto) {
-  std::fstream proto_stream(proto_path, std::ios::in | std::ios::binary);
+  std::filesystem::path proto_u8_path = std::filesystem::u8path(proto_path);
```
Comment (justinchuby) [inline onnx/common/file_utils.h:19]: Do we need to include `<filesystem>`?

**Comment by justinchuby** [comment]
Please follow https://github.com/onnx/onnx/pull/5806/checks?check_run_id=19631633187 to signoff your commits. Thank you for your contribution!

File: onnx/common/file_utils.h [hunk 2]
```diff
@@ -16,7 +16,8 @@ namespace ONNX_NAMESPACE {
...
+  std::fstream proto_stream(proto_u8_path, std::ios::in | std::ios::binary);
```
Comment (justinchuby) [inline onnx/common/file_utils.h:20]: Is there a unit test we can create for it?

File: onnx/test/checker_test.py [hunk 3]
```diff
@@ -1058,6 +1058,24 @@ def test_empty_list_attribute(self):
...
+        node_def = helper.make_node('Identity', ['input'], ['output'])
+        graph_def = helper.make_graph([node_def], 'simple_model', [input_tensor], [output_tensor])
+        model_def = helper.make_model(graph_def, producer_name='simple_onnx_generator')
+
+        self.unicode_model_path = '程序员.onnx'
+        onnx.save(model_def, self.unicode_model_path)
+
+        checker.check_model(self.unicode_model_path, True, True)
+
+    def tearDown(self):
```
Comment (justinchuby) [inline onnx/test/checker_test.py:1074]: I would use tempfile for creating a temporary folder. 

```
with tempfile.TemporaryDirectory() as temp_dir:
    unicode_model_path = os.path.join(temp_dir, "模型モデル모델✨.onnx")
```

File: onnx/test/checker_test.py [hunk 4]
```diff
@@ -1058,6 +1058,21 @@ def test_empty_list_attribute(self):
...
+        # create temp model file
+        input_tensor = helper.make_tensor_value_info('input', onnx.TensorProto.FLOAT, [1])
+        output_tensor = helper.make_tensor_value_info('output', onnx.TensorProto.FLOAT, [1])
+        node_def = helper.make_node('Identity', ['input'], ['output'])
+        graph_def = helper.make_graph([node_def], 'simple_model', [input_tensor], [output_tensor])
+        model_def = helper.make_model(graph_def, producer_name='simple_onnx_generator')
+
+        import tempfile
+        with tempfile.TemporaryDirectory() as temp_dir:
+            import os
```
Comment (justinchuby) [inline onnx/test/checker_test.py:1071]: Consider moving all imports to the top of the file

File: onnx/test/checker_test.py [hunk 5]
```diff
@@ -1058,6 +1058,21 @@ def test_empty_list_attribute(self):
...
 
+    def test_check_model_unicode_onnx_name(self) -> None:
+        # create temp model file
+        input_tensor = helper.make_tensor_value_info('input', onnx.TensorProto.FLOAT, [1])
+        output_tensor = helper.make_tensor_value_info('output', onnx.TensorProto.FLOAT, [1])
+        node_def = helper.make_node('Identity', ['input'], ['output'])
+        graph_def = helper.make_graph([node_def], 'simple_model', [input_tensor], [output_tensor])
+        model_def = helper.make_model(graph_def, producer_name='simple_onnx_generator')
+
+        import tempfile
```
Comment (justinchuby) [inline onnx/test/checker_test.py:1069]: Avoid non global imports

File: onnx/test/checker_test.py [hunk 6]
```diff
@@ -1058,6 +1058,21 @@ def test_empty_list_attribute(self):
...
+        with tempfile.TemporaryDirectory() as temp_dir:
+            import os
+            unicode_model_path = os.path.join(temp_dir, "程序员.onnx")
```
Comment (justinchuby) [inline onnx/test/checker_test.py:1072]: I suggest this name 

模型モデル모델✨.onnx

just to cover cjk and emoji

**Review by justinchuby — COMMENTED** [review]
Thanks! Final round of comments

**Comment by justinchuby** [comment]
@gramalingam for approval

**Pull request merged**